src/overlays/actors/ovl_En_Peehat/z_en_peehat.c

```diff
@@ -92,85 +99,987 @@ static ColliderJntSphInit D_80AD2790 = {
+    PEAHAT_STATE_STUNNED,
+    PEAHAT_STATE_SEEK_PLAYER,
+    PEAHAT_STATE_15
+} PeahatState;
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(targetArrowOffset, 700, ICHAIN_STOP),
+};
+
+void EnPeehat_SetupAction(EnPeehat* this, EnPeehatActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnPeehat_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnPeehat* this = THIS;
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    Actor_SetScale(thisx, 36.0f * 0.001f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_06001C80, &D_060009C4, this->jointTable, this->morphTable, 24);
+    ActorShape_Init(&this->actor.shape, 100.0f, ActorShadow_DrawCircle, 27.0f);
+    this->actor.focus.pos = this->actor.world.pos;
+    this->unk2D4 = 0;
+    this->actor.world.rot.y = 0;
+    this->actor.colChkInfo.mass = MASS_HEAVY;
+    this->actor.colChkInfo.health = 6;
+    this->actor.colChkInfo.damageTable = &sDamageTable;
+    this->actor.floorHeight = this->actor.world.pos.y;
+    Collider_InitCylinder(globalCtx, &this->colCylinder);
+    Collider_SetCylinder(globalCtx, &this->colCylinder, thisx, &sCylinderInit);
+    Collider_InitQuad(globalCtx, &this->colQuad);
+    Collider_SetQuad(globalCtx, &this->colQuad, thisx, &sQuadInit);
+    Collider_InitJntSph(globalCtx, &this->colJntSph);
+    Collider_SetJntSph(globalCtx, &this->colJntSph, thisx, &sJntSphInit, this->colJntSphItemList);
+
+    this->actor.naviEnemyId = 0x48;
+    this->xzDistToRise = 740.0f;
+    this->xzDistMax = 1200.0f;
+    this->actor.uncullZoneForward = 4000.0f;
+    this->actor.uncullZoneScale = 800.0f;
+    this->actor.uncullZoneDownward = 1800.0f;
+    switch (this->actor.params) {
+        case TYPE_GROUNDED:
+            EnPeehat_Ground_SetStateGround(this);
+            break;
+        case TYPE_FLYING:
+            this->actor.uncullZoneForward = 4200.0f;
+            this->xzDistToRise = 2800.0f;
+            this->xzDistMax = 1400.0f;
+            EnPeehat_Flying_SetStateGround(this);
+            this->actor.flags &= ~1;
+            break;
+        case TYPE_LARVA:
+            this->actor.scale.x = this->actor.scale.z = 0.006f;
+            this->actor.scale.y = 0.003f;
+            this->colCylinder.dim.radius = 25;
+            this->colCylinder.dim.height = 15;
+            this->colCylinder.dim.yShift = -5;
+            this->colCylinder.info.bumper.dmgFlags = 0x1F824;
+            this->colQuad.base.atFlags = AT_ON | AT_TYPE_ENEMY;
+            this->colQuad.base.acFlags = AC_ON | AC_TYPE_PLAYER;
+            this->actor.naviEnemyId = 0x49; // Larva
+            EnPeehat_Larva_SetStateSeekPlayer(this);
+    }
+}
+
+void EnPeehat_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnPeehat* this = THIS;
+    EnPeehat* parent;
+
+    Collider_DestroyCylinder(globalCtx, &this->colCylinder);
+    Collider_DestroyJntSph(globalCtx, &this->colJntSph);
+
+    // If larva, decrement total larva spawned
+    if (this->actor.params > 0) {
```

Comment:
can use your type enum throughout the file. in which case comment is prob not needed